web-common/src/features/entity-management/resource-invalidations.ts

```diff
@@ -1,226 +1,21 @@
-          resource.source?.spec?.sinkConnector ??
-          resource.model?.spec?.connector ??
-          "",
-      }),
-    );
-  }
-  fileArtifacts.updateArtifacts(resource);
-  const failed = !!resource.meta.reconcileError;
-
-  const name = resource.meta?.name?.name ?? "";
-  let table: string | undefined;
-  switch (resource.meta.name?.kind) {
-    case ResourceKind.Source:
-    case ResourceKind.Model:
-      table = resource.source?.state?.table ?? resource.model?.state?.table;
-      if (table && resource.meta.name?.name === table)
-        // make sure table is populated
-        return invalidateProfilingQueries(queryClient, name, failed);
-      break;
-
-    case ResourceKind.MetricsView:
-      return invalidateMetricsViewData(queryClient, name, failed);
-
-    case ResourceKind.Component:
-      return invalidateChartData(queryClient, name, failed);
-
-    case ResourceKind.Dashboard:
-    // TODO
-  }
-}
-
-function invalidateRemovedResource(
-  queryClient: QueryClient,
-  instanceId: string,
-  resource: V1Resource,
-) {
-  const name = resource.meta?.name?.name ?? "";
-  queryClient.removeQueries(
-    getRuntimeServiceGetResourceQueryKey(instanceId, {
-      "name.name": name,
-      "name.kind": resource.meta?.name?.kind,
-    }),
-  );
-  fileArtifacts.softDeleteResource(resource);
-  // cancel queries to make sure any pending requests are cancelled.
-  // There could still be some errors because of the race condition between a view/table deleted and we getting the event
-  switch (resource?.meta?.name?.kind) {
-    case ResourceKind.Source:
-    case ResourceKind.Model:
-      void queryClient.cancelQueries({
-        predicate: (query) => isProfilingQuery(query, name),
-      });
-      void queryClient.invalidateQueries(
-        getConnectorServiceOLAPListTablesQueryKey(),
-      );
-      break;
-    case ResourceKind.MetricsView:
-      void queryClient.cancelQueries({
-        predicate: (query) => invalidationForMetricsViewData(query, name),
-      });
-      break;
-  }
-}
-
-// We should not invalidate queries when resource is either queued or is running reconcile
-function shouldSkipResource(
-  queryClient: QueryClient,
-  instanceId: string,
-  res: V1Resource,
-) {
-  switch (res.meta?.reconcileStatus) {
-    case V1ReconcileStatus.RECONCILE_STATUS_UNSPECIFIED:
-      return true;
-
-    case V1ReconcileStatus.RECONCILE_STATUS_PENDING:
-    case V1ReconcileStatus.RECONCILE_STATUS_RUNNING:
-      void refreshResource(queryClient, instanceId, res);
-      fileArtifacts.updateReconciling(res);
-      return true;
-  }
-
-  return false;
-}
 
 export function refreshResource(
   queryClient: QueryClient,
   instanceId: string,
   res: V1Resource,
 ) {
-  return queryClient.resetQueries(
+  return queryClient.setQueryData(
     getRuntimeServiceGetResourceQueryKey(instanceId, {
       "name.name": res.meta?.name?.name,
       "name.kind": res.meta?.name?.kind,
     }),
+    {
+      resource: res,
+    },
   );
 }
```

Comment:
Given this function is just a wrapper around `setQueryData`, can we remove the wrapper and use `setQueryData` directly? Else it requires a reader to mentally map "refreshResource" -> `setQueryData`, which seems unnecessary.